pkg/agent/runner.go

```diff
@@ -322,6 +333,102 @@ func (r *Runner) Run(ctx context.Context, logger *zap.Logger, vmInfoUpdated util
 	}
 }
 
+func (r *Runner) reportScalingEvent(timestamp time.Time, currentCU, targetCU uint32) {
```

Comment:
Suggestion: use more consistent names, right now there is `Real` vs `Hypothetical` and `ScalingEvent` vs `DesiredScaling`. 

`reportRealScaling` and `reportHypotheticalScaling` would be fine, but it suggests as if hypothetical value cannot be real. Maybe `Actual` vs `Desired`?

Reply:
+1 on more consistent naming. Let me explain where I'm coming from; I want your thoughts:

IIRC, the thing I was trying to distinguish here is that the "hypothetical"/"desired" scaling events can go far beyond the endpoint limits, and also contain the fractional CU values for each "part" (cpu/mem/lfc). We also change these events if the _components_ change significantly, even if the overall scaling is still the same (so: it's not like each "hypothetical"/"desired" scaling event constitutes _actual_ scaling)

I think I opted not to go fully for "desired" scaling also because IIRC "desired" is used elsewhere in the autoscaler-agent to mean "the scaling value we should be working towards", and _is_ restricted by endpoint CU limits.

Thoughts? Happy to go with `Actual` vs `Desired` if you think that makes more sense.